admiral/pkg/clusters/dependency_handler.go

```diff
@@ -87,15 +87,17 @@ func isIdentityMeshEnabled(identity string, remoteRegistry *RemoteRegistry) bool
 	return false
 }
 
-func getDestinationsToBeProcessed(
+func getDestinationsToBeProcessed(eventType admiral.EventType,
 	updatedDependency *v1.Dependency, remoteRegistry *RemoteRegistry) ([]string, bool) {
 	updatedDestinations := make([]string, 0)
 	existingDestination := remoteRegistry.AdmiralCache.SourceToDestinations.Get(updatedDependency.Spec.Source)
-
 	var nonMeshEnabledExists bool
 	lookup := make(map[string]bool)
-	for _, dest := range existingDestination {
-		lookup[dest] = true
+	//if this is an update, build a look up table to process only the diff
+	if eventType == admiral.Update {
```

Comment:
Why was this added?

Reply:
`Add` event for dependency record was never processed. Meaning all calls to this method would assume dependency record update (get from cache, compare that with updated object). Adding this will make sure that Add event will consider all destinations to be processed.

Reply:
Cool, thanks!